### ISSUE TITLE
FR: Show more than 1 day of availability at a time

### ISSUE THREAD

**Issue by aspiers** [calcom/cal.com#4006]
### Is your proposal related to a problem?

I find it slightly more difficult than I think it needs to be when booking slots on someone else's cal.com link, because I can only see their availability one day at a time.  This means if I'm weighing up which is better out of some slots on (say) Tuesday and some other slots on Wednesday, I have to keep the available slots from one of those days in my memory while looking at the other day, because I can't see them both at the same time.  This creates some extra cognitive load which makes the UI a bit harder to use.

### Describe the solution you'd like

It would be great to be able to see three days of availability at a time.  Here's an example from another product:

![image](https://user-images.githubusercontent.com/100738/187092180-7bd2932d-d4a0-417c-a3bc-46f15c8c4811.png)

I would suggest that days with no availability are maybe skipped.  So for example if Wednesday is the first of the three days, but Thursday and Saturday have no availability, it would show Wednesday, Friday, and Sunday, rather than Wednesday/Thursday/Friday.  In this case probably there should be some kind of visual cue warning about the skipped days, otherwise the user might book a time on the second of the three days mistakenly thinking it was Thursday when it was actually Friday (even though of course the days would be labelled at the top of each column).

### Describe alternatives you've considered

- Show a different number of days at the same time, e.g. 2 days, or even a whole week.
- Allow the user to select exactly which days are displayed, e.g. by toggling checkboxes for each day.  But maybe that's overly complex.

**Comment by PeerRich** [comment]
we are planning to allow event-type creators to choose different layouts

**Comment by aspiers** [comment]
Perfect!

**Comment by aspiers** [comment]
For reference, here's another competitor which already provides this:

![image](https://user-images.githubusercontent.com/100738/197646540-5c539b82-fba4-4f0a-a731-bc627eb2a69f.png)

**Comment by PeerRich** [comment]
yeah we plan to soon offer 3 layouts to chose from and this will be one of them!

**Comment by PeerRich** [comment]
@Jaibles has some mockups for this. definitely planned

**Comment by Jaibles** [comment]
![CleanShot 2023-01-26 at 11.45.45@2x.png](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/68ddf0c0-7738-4092-8b44-8d3d54e83d97/f3690df3-75ca-4023-8e07-c4b7ed1c4671)

Designs not final but has been explored and will be something we work on once we have the Weekly view!

**Comment by PeerRich** [comment]
really love this

**Comment by aspiers** [comment]
Me too, looks great!!

**Comment by Jaibles** [comment]
Designs are ready for review here: 

[Column - Cal - New Features (Figma)](https://www.figma.com/file/UajFu4M1APhn2AywAEJkJr/Cal---New-Features?node-id=2585%3A132829&t=k66pu9sMzDhZOA2q-1)

**Comment by sean-brydon** [comment]
Hey @JeroenReumkens i was spying on your PR for the new atom is this something you included already? 

[https://github.com/calcom/cal.com/pull/6792/commits/32affd14cbfd56b7ef559922ac92f1385044ab3f](https://github.com/calcom/cal.com/pull/6792/commits/32affd14cbfd56b7ef559922ac92f1385044ab3f) This commit makes me think it is (Could be wrong)

**Comment by JeroenReumkens** [comment]
Yes, it is partially included @sean-brydon  — mainly making it possible in the new layout. There's still some tweaking that needs to be done after new booker atom is done, and styling needs to be finished / improved.

**Comment by ringods** [comment]
> ![CleanShot 2023-01-26 at 11.45.45@2x.png](https://camo.githubusercontent.com/850eb1c6548fcc9560711010a43ed784052c82475324ad7a5ef27fb640264cb4/68747470733a2f2f75706c6f6164732e6c696e6561722e6170702f65383662663935372d643832662d343635652d623230352d3133353535396634623632332f36386464663063302d373733382d343039322d386234342d3864336435346538336439372f66333639306466332d373563612d343032332d386530372d633462376564316334363731)
> 
> Designs not final but has been explored and will be something we work on once we have the Weekly view!

I hope this is not the conceptual design which will land. If you show a multi day view (3 day, 5 day workweek, full week), make sure that you show a proper calendar where e.g. 1PM for every day is aligned. 

My proposal: show a proper week calendar view, gray out the zones which are not possible for the scheduler to select. Allow to select a block based on how I configured the scheduling link. If I allow for 60 minute blocks and the scheduler can't overrule the duration, only allow to plan 60 minute blocks.

The best I can compare it with is [SavvyCal week view](https://docs.savvycal.com/article/31-customizing-the-earliest-visible-week).

**Comment by Jaibles** [comment]
> I hope this is not the conceptual design which will land. If you show a multi day view (3 day, 5 day workweek, full week), make sure that you show a proper calendar where e.g. 1PM for every day is aligned. 


The screenshot in your comment is the column view. We also have weekly view in the development which behaves as you describe @ringods :)

**Comment by xeruf** [comment]
Appreciate the progress, but the previous design has arguments for it as well.
What if there are slots at 10 and 18? Lots of wasted screen estate and lots of scrolling.
I hate scrolling up and down in calendars just to see if there is something else.
If both are available, great!

**Comment by Jaibles** [comment]
> If both are available, great!

Both of these options will be available in addition to the current monthly view. You can choose which options to make available to your bookers. If you make multiple available, your bookers can choose which they prefer.
